apps/src/templates/manageStudents/ManageStudentsTable.jsx

```diff
@@ -729,6 +747,31 @@ class ManageStudentsTable extends Component {
               )}
             </div>
           )}
+          {noSectionCode.includes(loginType) && (
+            <div style={styles.sectionCodeBox}>
+              {i18n.sectionCodeWithColon()}
+              <strong>
+                {` ${i18n.notApplicable()} `}
+                <a onClick={() => this.showSectionCodeDialog()}>
+                  {i18n.whyWithQuestionMark()}
+                </a>
+              </strong>
+            </div>
+          )}
+          <div>
+            <BaseDialog
```

Comment:
Using a BaseDialog seems like a detail that NoSectionCodeDialog should know rather than a detail ManageStudentsTable should know. Would it be possible to have ManageStudentsTable render the NoSectionCodeDialog and have NoSectionCodeDialog render BaseDialog?

Reply:
Yes, I had the same though after I pushed the code late last night. BaseDialog definitely belongs under the domain of NoSectionCodeDialog. I refactored the code to reflect these changes. I now pass down the showSectionCodeDialog variable in stat as props to NoSectionCodeDialog. I already passed down the event handler for closing the dialog, so no need to make any changes there.

Reply:
Saw the new changes - they look great!